phygnn/layers/custom_layers.py

```diff
@@ -1527,23 +1539,42 @@ def call(self, x, hi_res_feature=None):
         hi_res_feature : tf.Tensor | np.ndarray
             This should be a 4D array for spatial enhancement model or 5D array
             for a spatiotemporal enhancement model (obs, spatial_1, spatial_2,
-            (temporal), 1). This is NaN where there are no observations and
-            real values where observations exist.
+            (temporal), features). This is NaN where there are no observations
+            and real values where observations exist.
+        exo_data : tf.Tensor | np.ndarray
+            This is an array of exogenous data used to imform the embedding,
+            like topography
 
         Returns
         -------
         x : tf.Tensor
-            Output tensor with embedded hi_res_feature and observation mask
-            concatenated to input.
+            Output tensor with embedding concatenated to input.
         """
         if hi_res_feature is None:
-            hi_res_feature = tf.fill(x[..., :1].shape, np.nan)
+            hr_shape = (*x[..., 0].shape, len(self.features))
+            hi_res_feature = tf.constant(np.nan, shape=hr_shape, dtype=x.dtype)
+
+        if exo_data is None and len(self.exo_features) > 0:
+            exo_shape = (*x[..., 0].shape, len(self.exo_features))
+            exo_data = tf.constant(0, shape=exo_shape, dtype=x.dtype)
+
+        if self.fill_method is None:
```

Comment:
This option for `fill_method=None` is not described in the docstring yet it is supported here. Can you describe or remove? Seems like you should remove this based on the assert statement in the init

Reply:
Yeah this should be kept actually, so the hidden_layers = None reduces to Sup3rConcatObs.